### PR TITLE
Prerequisites for Dynamic Process Support

### PR DESCRIPTION
**Pull request by yfguo**
These are fixes and refactoring for the dynamic process branch to work. Submitted for testing and merging.

### CODE REVIEW

**Comment by yfguo** [comment]
test:ch4/all

**Comment by yfguo** [comment]
test:jenkins/ch4/all

**Comment by yfguo** [comment]
test:jenkins/ch4/all

**Comment by yfguo** [comment]
test:jenkins/ch4/all

File: src/mpid/ch4/src/ch4r_proc.h [hunk 1]
```diff
@@ -482,14 +501,28 @@ static inline int MPIDIU_avt_release_ref(int avtid)
...
+                      mmap(NULL, MPIDI_CH4_Global.avt_mgr.mmaped_size,
+                           PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+    if (MPIDI_av_table == MAP_FAILED) {
+        abort();
+    }
+    MPIDI_CH4_Global.node_map = (MPID_Node_id_t **)
+                                mmap(NULL, MPIDI_CH4_Global.avt_mgr.mmaped_size,
+                                     PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+    if (MPIDI_CH4_Global.node_map == MAP_FAILED) {
+        abort();
```
Comment (hoopoepg) [inline src/mpid/ch4/src/ch4r_proc.h]: may be add some diagnostic message here?
just 'abort' is not user friendly :)

Comment (raffenet) [inline src/mpid/ch4/src/ch4r_proc.h]: Agreed. We should use the standard errcheck macros that eventually call `MPID_Abort`.

Comment (yfguo) [inline src/mpid/ch4/src/ch4r_proc.h]: Sure. I can fix that.

File: src/mpid/ch4/src/ch4_init.h [hunk 2]
```diff
@@ -510,57 +510,6 @@ MPL_STATIC_INLINE_PREFIX int MPID_Get_max_node_id(MPIR_Comm * comm, MPID_Node_id
...
-    }
-
-  fn_exit:
-    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_UPIDS_TO_LUPIDS);
-    return mpi_errno;
-  fn_fail:
-    goto fn_exit;
-}
-
-#undef FUNCNAME
```
Comment (wesbland) [inline src/mpid/ch4/src/ch4_init.h]: Should this chunk be part of the previous commit?

File: src/mpid/ch4/netmod/ofi/ofi_init.h [hunk 3]
```diff
@@ -993,10 +993,9 @@ static inline int MPIDI_NM_get_local_upids(MPIR_Comm * comm, size_t ** local_upi
     goto fn_exit;
 }
 
-static inline int MPIDI_OFI_upids_to_lupids_general(int size,
-                                                    size_t * remote_upid_size,
-                                                    char *remote_upids,
-                                                    int **remote_lupids, int use_av_table)
+static inline int MPIDI_NM_upids_to_lupids(int size,
+                                           size_t * remote_upid_size,
+                                           char *remote_upids, int **remote_lupids)
```
Comment (wesbland) [inline src/mpid/ch4/netmod/ofi/ofi_init.h]: The rename can probably go with the first commit that moves this function to a different file.

File: src/mpid/ch4/netmod/ofi/ofi_init.h [hunk 4]
```diff
@@ -1072,14 +1067,6 @@ static inline int MPIDI_OFI_upids_to_lupids_general(int size,
...
 }
 
-static inline int MPIDI_NM_upids_to_lupids(int size,
-                                           size_t * remote_upid_size,
-                                           char *remote_upids, int **remote_lupids)
-{
-    return MPIDI_OFI_upids_to_lupids_general(size, remote_upid_size, remote_upids,
-                                             remote_lupids, MPIDI_Global.settings.enable_av_table);
-}
-
```
Comment (wesbland) [inline src/mpid/ch4/netmod/ofi/ofi_init.h]: Should this get combined with the first commit too?

File: src/mpid/ch4/netmod/ofi/ofi_init.h (hunk 3 above)
Comment (yfguo) [inline src/mpid/ch4/netmod/ofi/ofi_init.h]: Actually, this change is related to the current commit. The original `MPIDI_OFI_upids_to_lupids_general` use different code paths for `FI_AV_MAP` and `FI_AV_TABLE`. Therefore it needs the `MPIDI_NM_upids_to_lupids` as a wrapper for the ADI.

As a result of this commit, both modes can use the exactly same code path. Therefore, the wrapper should go away. So I figured the renaming should be in this commit as well.

File: src/mpid/ch4/src/ch4_init.h (hunk 2 above)
Comment (yfguo) [inline src/mpid/ch4/src/ch4_init.h]: You are right. I messed up when splitting the commits.

**Comment by yfguo** [comment]
I have pushed the fixes for the comments. Testing again.

test:jenkins/all/ch4

File: src/mpid/ch4/include/mpidpre.h [hunk 5]
```diff
@@ -440,13 +440,15 @@ extern MPIDI_av_table_t *MPIDI_av_table0;
...
+/*
+ * operation for (avtid, lpid) to/from "lupid"
+ * 1 bit is reserved for "new_avt_mark". It will be cleared before accessing
+ * the avtid and lpid. Therefore, the avtid mask does have that bit set to 0
+ */
+#define MPIDIU_AVTID_BITS                    (7)
+#define MPIDIU_LPID_BITS                     (8 * sizeof(int) - (MPIDIU_AVTID_BITS + 1))
+#define MPIDIU_LPID_MASK                     (0xFFFFFFFFU >> (MPIDIU_AVTID_BITS + 1))
+#define MPIDIU_AVTID_MASK                    (~MPIDIU_LPID_MASK)
 #define MPIDIU_NEW_AVT_MARK                  (0x80000000U)
```
Comment (raffenet) [inline src/mpid/ch4/include/mpidpre.h]: The changes in this commit are all loosely related, making it kind of hard to hollow. Can you split into 3 different commits?

Comment (yfguo) [inline src/mpid/ch4/include/mpidpre.h]: I have split this one into three commits and fix the typo of `ch4r_comm.sh`.

**Comment by yfguo** [comment]
test:jenkins/ch4/all

**Comment by wesbland** [comment]
@raffenet - Looks like this has the changes you were looking for. Do we need to do the file rename for this? Given that Pavan is pushing for the paper, I can take over for @yfguo if necessary here. I'd rather not wait for two weeks to get this stuff moving.

**Comment by wesbland** [comment]
test:jenkins/ch4/all

**Comment by wesbland** [comment]
test:jenkins/ch4/all

**Comment by raffenet** [comment]
test:jenkins/ch4/most

**Comment by raffenet** [comment]
I added 2 fixup commits that should address my only comments. Testing again:
test:jenkins/ch4/ofi
test:jenkins/ch4/ucx
